lib/parsers/helpers.ex

```diff
@@ -62,4 +62,56 @@ defmodule ICalex.Parsers.Helpers do
+  CALSCALE:GREGORIAN\r\nMETHOD:REQUEST\r\nBEGIN:VEVENT\r\nDTSTART:20220601T043000Z\r\nDTEND:20220601T050000Z\r\n
+  DTSTAMP:20220531T013709Z\r\nORGANIZER;CN=kimhoang@tsc.ai:mailto:someone@tsc.ai\r\n
+  UID:040000008200E00074C5B7101A82E008000000004A92673A0974D801000000000000000\r\n 0100000003AA8F50FFD17984585F51196380DC9C6\r\n
+  ATTENDEE;CUTYPE=INDIVIDUAL;ROLE=REQ-PARTICIPANT;PARTSTAT=NEEDS-ACTION;RSVP=\r\n TRUE;CN=12345c@inbound.postmarkapp.com;X-NUM-GUES\r\n
+  TS=0:mailto:12345c@inbound.postmarkapp.com\r\nX-MICROSOFT-CDO-OWNERAPPTID:-674825255\r\nCREATED:20220530T093959Z\r\n
+  DESCRIPTION:-::~:~::~:~:~:~:~:~:~:~:~:~:~:~:~:~:~:~:~:~:~:~:~:~:~:~:~:~:~:~\r\n :~:~:~:~:~:~:~:~::~:~::-\\nDo not edit this section of the description.\\n\\nV\r\n
+  iew your event at https://calendar.google.com/calendar/event?action=VIEW&ei\r\n d=XzYwcTMwYzFnNjBvNzBlMWk2MG80YWMxZzYwcmo4Z3BsODhyajJjMWg4NHMzNGg5ZzYwczMwY\r\n
+  zFnNjBvMzBjMWc2aDBqaWNobTZzcGsyYzFwNnNxNDhlMWc2NG8zMGMxZzYwbzMwYzFnNjBvMzBj\r\n MWc2MG8zMmMxZzYwbzMwYzFnNmQwazJlMjY2a280Y2hpNDY0cmppZTFrNmtzM2FoaGw2NG9qaWR\r\n
+  oajcwbzQ4Z3BwOGNyMCAwOTBkZWRiN2JlZWMwNTBhNzAzNWVlNmExMzY1MDc0Y1BpbmJvdW5kLn\r\n Bvc3RtYXJrYXBwLmNvbQ&tok=MTUja2ltaG9hbmdAdHNjLmFpMjE3YjIyZmEwODY2OTU1MWM0Yj\r\n
+  FhYWFjMTM1ZTBiNmM5ODAyZWE3ZQ&ctz=Asia%2FSingapore&hl=en&es=0.\\n-::~:~::~:~:\r\n ~:~:~:~:~:~:~:~:~:~:~:~:~:~:~:~:~:~:~:~:~:~:~:~:~:~:~:~:~:~:~:~:~:~::~:~::-\r\n
+  LAST-MODIFIED:20220531T013708Z\r\nLOCATION:\r\nSEQUENCE:2\r\nSTATUS:CONFIRMED\r\nSUMMARY:Test Webhook Local\r\nTRANSP:OPAQUE\r\nBEGIN:VALARM\r\nACTION:DISPLAY\r\n
+  DESCRIPTION:This is an event reminder\r\nTRIGGER:-P0DT0H15M0S\r\nEND:VALARM\r\nEND:VEVENT\r\nEND:VCALENDAR\r\n"
+
+  In the field UID there is a linebreak \r\n:
+  UID:040000008200E00074C5B7101A82E008000000004A92673A0974D801000000000000000\r\n 0100000003AA8F50FFD17984585F51196380DC9C6
+  Which, when going throught String.split("\r\n") will become a list ["UID:040000008200E00074C5B7101A82E008000000004A92673A0974D801000000000000000",
+  " 0100000003AA8F50FFD17984585F51196380DC9C6"]. And when this list go throught parser(nil), the value " 0100000003AA8F50FFD17984585F51196380DC9C6" will
+  cause error: (MatchError) no match of right hand side value: [" 0100000003AA8F50FFD17984585F51196380DC9C6"]
+  Basically the second value in the list is not a legic ical field, it was created because there is an incorrect linebreak in the original data that cause
+  the UID field to be cut into two, with one is still a ablegit field with key - value structure and another one is not.
+
+  This update is to clean up such incorrect linebreaks inside each field.
+  """
+  def fix_linebreak_in_the_middle_of_value(components) when is_list(components) do
+    regex = ~r/^[A-Z-]*[:;][A-Z0-9]*/
+    IO.inspect(components)
+    fix_linebreak_in_the_middle_of_value(components, [], regex)
+  end
+
+  def fix_linebreak_in_the_middle_of_value([head | tail], cleaned_components, regex)
```

Comment:
In general, I think this code works well, however in idiomatic Elixir I don't see a need for a recursive function. Essentially you want to combine two elements in a list, which sounds perfect for a Enum.reduce.

```
 Enum.reduce(components, [], fn component, acc ->
      if String.match?(component, regex) do
        acc ++ [component] # unoptimised
      else
        last_added = Enum.at(acc, -1)
        List.replace_at(acc, -1, last_added <> component)
      end
    end)
```

Reply:
You can keep the code as it is BTW, no need to change to mine, its just my observation.